### PR TITLE
RHINENG-9505: register system within candlepin

### PR DESCRIPTION
**Pull request by MichaelMraka**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: manager/controllers/template_systems_update.go [hunk 1]
```diff
@@ -155,3 +168,63 @@ func templateArchVersionMatch(
...
+		Environments: environments,
+	}
+	for _, consumer := range consumers {
+		if consumer.Consumer == nil {
+			err = errors2.Join(err, errors.Errorf("Missing owner_id for '%s'", consumer.InventoryID))
+			continue
+		}
+		resp, apiErr := callCandlepin(c, *consumer.Consumer, &updateReq)
+		// check response
+		if err != nil {
```
Comment (psegedy) [inline manager/controllers/template_systems_update.go:224]:
```suggestion
		if apiErr != nil {
```

**Review by psegedy — CHANGES_REQUESTED** [review]

File: manager/controllers/template_systems_update.go (hunk 1 above)
Comment (MichaelMraka) [reply manager/controllers/template_systems_update.go]: 👍🏼

File: manager/middlewares/rbac.go [hunk 2]
```diff
@@ -32,6 +32,8 @@ var granularPerms = map[string]string{
 	"BaselineUpdateHandler":        "patch:template:write",
 	"BaselineDeleteHandler":        "patch:template:write",
 	"BaselineSystemsRemoveHandler": "patch:template:write",
+	"TemplateSystemsUpdateHandler": "patch:template:write",
+	"TemplateSystemsDeleteHandler": "patch:template:write",
```
Comment (jlsherrill) [inline manager/middlewares/rbac.go:36]: Should this be  ```content-sources:templates:write``` ?

https://github.com/RedHatInsights/rbac-config/blob/master/configs/stage/roles/content-sources.json#L61

Comment (MichaelMraka) [reply manager/middlewares/rbac.go]: good point

**Pull request merged**